feat(RotationalTeleport): teleport with player reorientation

thacio opened the pull request:
Include teleport system which allows view rotation. The pointer which rotates must be a VRTK_SimplePointer. The teleport controller can be any type inherited of VRTK_WorldPointer.

thestonefox commented:
I'll check this out in the morning :)

Is it ok to close the other PR now? https://github.com/thestonefox/SteamVR_Unity_Toolkit/pull/312

thacio commented:
Yup! It's updated with the latest commit.

thestonefox commented on Assets/SteamVR_Unity_Toolkit/Scripts/Abstractions/VRTK_WorldPointer.cs:
```diff
@@ -283,6 +289,9 @@ private void TurnOnBeam(uint index)
                 isActive = true;
                 destinationSetActive = true;
+
+                if (OnPointerOn != null)
```
curly braces around the conditional block

thestonefox commented on Assets/SteamVR_Unity_Toolkit/Scripts/Abstractions/VRTK_WorldPointer.cs:
```diff
@@ -293,8 +302,12 @@ private void TurnOffBeam(uint index)
                 TogglePointer(false);
                 isActive = false;
+
+                if (OnPointerOff != null)
```
curly whirly braces again :)

thestonefox commented on Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_SimplePointer.cs:
```diff
@@ -22,6 +22,8 @@ public class VRTK_SimplePointer : VRTK_WorldPointer
         public bool showPointerTip = true;
         public GameObject customPointerCursor;
+        [Tooltip("Check this if you would like to keep the custom pointer cursor material.")]
+        public bool keepCustomPointerMaterial = false;
```
is it worth moving this into WorldPointer as it seems both SimplePointer and BezierPointer have it (and they both extend world pointer anyway)

thacio replied:
customPointerCursor is implemented by them, instead of WorldPointer, to they probably should stick together.
I think it would be better to leave there for now and move customPointerCursor and keepCustomPointerMaterial to WorldPointer in another commit.

thacio commented:
Done. Updated the example scene with instructions how to use the feature and changed the direction pointer to something that kind of maybe from far away looks like an arrow.
